pandas/core/indexing.py

```diff
@@ -2407,9 +2407,14 @@ def pred(part) -> bool:
         """
         # true when slice does *not* reduce, False when part is a tuple,
         # i.e. MultiIndex slice
-        return (isinstance(part, slice) or is_list_like(part)) and not isinstance(
-            part, tuple
-        )
+        if isinstance(part, tuple):
+            # check for sub-slice:
```

Comment:
can you add "GH#39421" here

instead of a for loop can you do this with an `any(...)`